### PR TITLE
Misc size regression script improvements.

### PR DESCRIPTION
**Pull request by tzarc**
## Description

- Sets environment variable SIZE_REGRESSION_EXECUTING during execution
  so hook scripts like `post-checkout` may skip processing.
- Forces checkout of the target branch, including removal of all
  temporary object files in the process.
- Prints out a warning on exit stating that the git repository is in an
  indeterminate state, and the user needs to swap back to whatever
  intended branch they were working with originally.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by KarlK90 — APPROVED** [review]
LGTM!

**Review by fauxpark — APPROVED** [review]

File: util/size_regression.sh [hunk 1]
```diff
@@ -49,7 +66,9 @@ function build_executor() {
         revision=$(echo $line | cut -d' ' -f1)
 
         make distclean >/dev/null 2>&1
-        git checkout $revision >/dev/null 2>&1 || { echo "Failed to check out revision ${revision}" >&2 ; exit 1 ; }
+
+        git clean -xfd >/dev/null 2>&1 || true
```
Comment (zvecr) [inline util/size_regression.sh:70]: The downside here is that it will remove items from global ignore, in my case blow away my direnv config

Comment (tzarc) [reply util/size_regression.sh:70]: Removed.

**Comment by zvecr** [comment]
Maybe the git clean part could be hidden behind a `-f` argument?

**Pull request merged**